client/driver/docker.go

```diff
@@ -254,11 +254,25 @@ func (d *DockerDriver) dockerClients() (*docker.Client, *docker.Client, error) {
 			if cert+key+ca != "" {
 				d.logger.Printf("[DEBUG] driver.docker: using TLS client connection to %s", dockerEndpoint)
 				client, err = docker.NewTLSClient(dockerEndpoint, cert, key, ca)
+				if err != nil {
+					merr.Errors = append(merr.Errors, err)
+				}
+				waitClient, err = docker.NewTLSClient(dockerEndpoint, cert, key, ca)
+				if err != nil {
+					merr.Errors = append(merr.Errors, err)
+				}
 			} else {
 				d.logger.Printf("[DEBUG] driver.docker: using standard client connection to %s", dockerEndpoint)
 				client, err = docker.NewClient(dockerEndpoint)
+				if err != nil {
+					merr.Errors = append(merr.Errors, err)
+				}
+				waitClient, err = docker.NewClient(dockerEndpoint)
+				if err != nil {
+					merr.Errors = append(merr.Errors, err)
+				}
 			}
-			client.HTTPClient.Timeout = dockerTimeout
+			client.SetTimeout(dockerTimeout)
 			return
 		}
 
```

Comment:
Need to set the timeout  in the case they don't set the docker endpoint too: https://github.com/hashicorp/nomad/blob/4e283d6c0ecdfc549d510a65d8a5ce77e12928d8/client/driver/docker.go#L284